src/main/java/com/flagsmith/flagengine/Engine.java

```diff
@@ -1,153 +1,111 @@
 package com.flagsmith.flagengine;
 
-import com.flagsmith.flagengine.environments.EnvironmentModel;
-import com.flagsmith.flagengine.features.FeatureModel;
-import com.flagsmith.flagengine.features.FeatureStateModel;
-import com.flagsmith.flagengine.identities.IdentityModel;
-import com.flagsmith.flagengine.identities.traits.TraitModel;
 import com.flagsmith.flagengine.segments.SegmentEvaluator;
-import com.flagsmith.flagengine.segments.SegmentModel;
-import com.flagsmith.flagengine.utils.exceptions.FeatureStateNotFound;
+import com.flagsmith.flagengine.utils.Hashing;
+import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
-import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.ImmutablePair;
 
 public class Engine {
-
   /**
-   * Get a list of feature states for a given environment.
+   * Get evaluation result for a given evaluation context.
    *
-   * @param environment Instance of the Environment.
+   * @param context Evaluation context.
+   * @return Evaluation result.
    */
-  public static List<FeatureStateModel> getEnvironmentFeatureStates(EnvironmentModel environment) {
-    if (environment.getProject().getHideDisabledFlags()) {
-      return environment.getFeatureStates()
-          .stream()
-          .filter((featureState) -> featureState.getEnabled())
-          .collect(Collectors.toList());
+  public static EvaluationResult getEvaluationResult(EvaluationContext context) {
+    List<SegmentResult> segments = new ArrayList<>();
+    HashMap<String, ImmutablePair<String, FeatureContext>> segmentFeatureContexts = new HashMap<>();
+    List<FlagResult> flags = new ArrayList<>();
+
+    for (SegmentContext segmentContext : context.getSegments().getAdditionalProperties().values()) {
```

Comment:
NIT: Ideally we could abstract the different parts more. To me it would make SDK synchronization easier with a mental schema being that getEvaluationResult:
- Evaluate segments (getIdentitySegments and build the segment overrides)
- Evaluate features using the segments